protoc-gen-grpc-gateway/main.go

```diff
@@ -44,12 +45,20 @@ var (
 	date    = "unknown"
 )
 
+func buildVersion() string {
+	v, ok := debug.ReadBuildInfo()
+	if !ok {
+		return version
+	}
+	return v.Main.Version
+}
+
 func main() {
 	flag.Parse()
 	defer glog.Flush()
 
 	if *versionFlag {
-		fmt.Printf("Version %v, commit %v, built at %v\n", version, commit, date)
+		fmt.Printf("Version %v, commit %v, built at %v\n", buildVersion(), commit, date)
```

Comment:
Could you add the 1.18 specific stuff to a `main_go1.18` and support the old format for Go 1.17 and below? We try to support the last 3 releases I think.

Reply:
done, if i understood you correctly